src/triple.rs

```diff
@@ -127,6 +127,35 @@ impl Triple {
             _ => return Err(()),
         })
     }
+
+    /// The C data model for a given target. If the model is not known, returns `Err(())`.
+    pub fn data_model(&self) -> Result<CDataModel, ()> {
+        let pointer_width = self.pointer_width()?;
+        if pointer_width == PointerWidth::U64 {
+            if self.operating_system == OperatingSystem::Windows {
+                Ok(CDataModel::LLP64)
+            } else if self.default_calling_convention() == Ok(CallingConvention::SystemV)
+                || self.architecture == Architecture::Wasm64 {
+                Ok(CDataModel::LP64)
+            } else {
+                Err(())
+            }
+        } else if pointer_width == PointerWidth::U32 {
+            if self.operating_system == OperatingSystem::Windows
+                || self.default_calling_convention() == Ok(CallingConvention::SystemV)
+                || self.architecture == Architecture::Wasm32 {
+                Ok(CDataModel::ILP32)
+            } else {
+                Err(())
+            }
+        } else {
```

Comment:
This `else` is for `PointerWIdth::U16`? Could you switch the if-else chain here to a `match` so that it's more explicit?

Also, I'm confused about why a 16-bit pointer system uses a data model with `P32` in it.

Reply:
I'm not sure why it uses 32-bit pointers either, but all the sources I've found seem to back that up. I haven't found anything official, though.

I'll see if I can make the code a little more readable with a `match`.

- https://en.cppreference.com/w/cpp/language/types
- http://nickdesaulniers.github.io/blog/2016/05/30/data-models-and-word-size/
- http://dewkumar.blogspot.com/2012/03/cc-data-model-3264-bit.html